### ISSUE TITLE
Adding updated notes/comments to the actual documentation pages in the guides

### ISSUE THREAD

**Issue by CoolColin** [wet-boew/wet-boew#7540]
As a member of the AWG I am writing on behalf of the entire community as I am trying to determine who to talk to or who to coordinate with in order to get "Accessibility Notes" and "Best Practice" notes added to the documentation pages on some of the widgets and extra functional components of the WET Toolkit which are deemed accessible yet have been tested and proven by my peers that some issues have been noted? 

For example we have discovered issues with the expand/collapse plug-in for the details and summary elements. (http://wet-boew.github.io/v4.0-ci/demos/details/details-en.html)  We would love to add a note to this page regarding our findings in order to make developers aware that there are in fact minor accessibility issues with this polyfill. We would also like to possibly add “Best Practices” or “Best used for” notes as well. 

Another example of a component we have documented issues with is the Datalist polyfill. http://wet-boew.github.io/v4.0-ci/demos/datalist/datalist-en.html  

For all working with these guides and examples I ask:

1. Would you find these notes beneficial?
2. Would other individuals want to share or add similar comments to these pages as well as we further discover hiccups or bugs?

Many Thanks…

**Comment by pjackson28** [comment]
@CoolColin Here are my two cents on the subject (just an opinion):

1. For your suggestion for best used for, I can see value in guidance along those lines as it seems some components are being used in the wrong scenarios. There are some components that already have such guidance (such as [Toggle](http://wet-boew.github.io/wet-boew/docs/ref/toggle/toggle-en.html) with Use When and Don't Use When) so it would be good to see that guidance extended across all documentation. It likely would be a benefit to all users if the components are only used in the appropriate scenarios. I suggest following the approach in the Toggle documentation to keep consistent with how guidance is conveyed.

2. One challenge with what you describe is that the source of accessibility issues is not always WET. For instance with the datalist polyfill, it is natively supported by most browsers so the issues that you would be encountering is an internal issue to a browser and/or a screen reader rather than WET since in most cases the WET polyfill isn't even loaded. In that case it is an external HTML5 support issue and not something that WET can control so wouldn't seem to be appropriate for the WET documentation.

Same with certain bugs we've seen where WET is following WCAG 2.0, WAI-ARIA and HTML5 and it works perfectly fine in certain screen readers but not perfectly in certain others. This particularly is apparent in comparing the results in NVDA versus Jaws or even different versions of Jaws (e.g., the handling of role="application", non-h1 elements in section elements, role="dialog", etc). There is uneven support across screen readers (like there is across browsers) so I'm not sure WET documentation would be the best place to be addressing browser/screen reader support issues for HTML5 and WAI-ARIA (or any other issues that are not exclusive to WET).

All this should be documented in the issue tracker, and the issues and discussion seem to be really good there, but I'm not sure what value there would be in duplicating it in the documentation.

3. As for your suggestion about cross-posting issues filed in the issue tracker to the documentation (following up on the previous point), that would mean duplicate effort maintaining updates about a given issue in two different places. This could mean that an issue could be identified as fixed in the issue tracker yet still identified as an issue in the documentation (if maintainers forget to update both places at the same time, which is quite likely to happen). It could also lessen the importance of other issues where there could be a severe mobile support bug for a given component that is only in the issue tracker but a minor accessibility/usability issue is highlighted in the documentation.

4. As for your suggestion about having comments no the documentation, I don't think that is the appropriate venue for comments or discussion. The issue tracker is much better suited for that purpose as it is specifically designed for comments and discussion (unlike the documentation where it likely would not be possible to implement that functionality there).

**Comment by dorina1** [comment]
The developers are using the code from the working examples on the assumption that they are fully accessible, where some are having issues especially when using the IE 11 browser. The documentation is not very specific and quite general.

For example the [datalist polyfill](http://wet-boew.github.io/v4.0-ci/docs/ref/datalist/datalist-en.html) you mentioned, in the documentation is mentioned "Because some browsers do not support this functionality natively, this polyfill emulates the same functionality using generic HTML and WAI-ARIA.". 

When we check the browsers for the support, IE 11 is listed as supporting this functionality, however it does not work with JAWS. It works perfectly well with JAWS and Firefox. How are we suppose to know that? Is there a compiled list of accessibility issues reported for all working examples?

The issue tracker is fine, but for example AWG reported the issue with the [Form validation: "required" spoken twice for screen readers](https://github.com/wet-boew/wet-boew/issues/7002) in 2015 and one year after we still have the same problem.

Many thanks.
@wet-boew/access-working-group

**Comment by pjackson28** [comment]
@dorina1 The problem that I was raising is that the issue in question isn't with the datalist polyfill that WET provides (which doesn't execute in IE11), it is with the native support for the HTML5 datalist element in the IE11 + Jaws combination. The issue exists regardless of whether or not WET is implemented because it is an issue with how IE11 and/or JAWS implements the HTML5 specification. It is similar to the Jaws bug where the heading level is incremented incorrectly when the HTML5 section element is used with a heading element. So if you nest an h2 in a section element (very common HTML5 pattern) then Jaws would read it as a level 3 heading (whereas it is read perfectly fine in NVDA). Both the HTML5 datalist and section/heading issues happen in JAWS regardless of whether or not WET is used and they can't be fixed by WET as they would have to be fixed by the respective vendors. That is why I was questioning why we should be including in WET documentation vendor product bugs that are independent of WET.

As for the "required" being spoken twice, the likely reason that it hasn't been fixed is that it will negatively impact other users by removing it. If you take off required="required" then you break form validation (since the field is no longer identified to the plugin as being required) and if you take "required" out of the label then you've removed the required indicator that some users rely upon. Both of those end results would be far more detrimental than hearing "required" twice.

**Comment by dorina1** [comment]
@pjackson28 Thank you so much for the clarifications. I did not realize that required="required" is used for the form validation. I have also contacted Freedom Scientific for fixes required with JAWS functionality and ARIA, HTML 5.

Maybe another idea will be for each GoC department to compile a list of "known issues" based on the browser version and adaptive technology version certified to be used combination and go from there?
@wet-boew/access-working-group

**Comment by pjackson28** [comment]
@dorina1 I think there being a collective list of known issues with browser and adaptive technology combinations could be quite useful, particularly if it is used to ensure those issues are filed with the respective vendors and to follow up with those vendors. There are likely some issues that either haven't been filed yet or that the vendors haven't taken action on, despite the best efforts of people like @dfarough who have been very good about filing issues with vendors.

**Comment by nschonni** [comment]
Less about specific browser/AT combos, but http://html5accessibility.com/ shows the accessibility support and issues in the major browsers for HTML5 features.

**Comment by mysticjeff** [comment]
Yes, we reference the site.
However, it primarily looks at support at the browser level and not the Adaptive Technology level.  It also focuses on screen reader support and there are about 130 different adaptive technology software  packages in use in the federal public service alone, covering vision, hearing, physical/injury and cognitive / learning style enhancement technology.


-----Original Message-----

**Comment by hsrudnicki** [comment]
Please update again the status of this, whether this remains still an issue or can be this closed?
As well shouldn't be this above encompassed by the Content: Documentation issue #5574 ?

**Comment by CoolColin** [comment]
I believe for the time being we will close this ticket. With come current work being done on the style guide and discussions the AWG is having with TBS around publishing policies/guidelines etc... I would rather close this for now and think of a new direction in the future as this may not be as sustainable as I had envisioned.
